lib/sort_by_length.rb

```diff
@@ -1,7 +1,40 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(nlog(n)) for merge sort - although bubble sort and insertion sort can have a best-case time complexity of O(n)
+# they have the worst case time complexity of O(n^2) (which is why I chose to do a merge sort).
+# Space complexity: O(n)
 def sort_by_length(my_sentence)
```

Comment:
nice work at a MergeSort.  Just be aware doing things like `right[1..right.length]` are O(n) operations because they copy most of the given array... so Your solution is actually O(n^2 log n).  

You can do better just by keeping indexes for the left and right in your merge method.  

That said, this works, well done.